components/places/src/bookmark_sync/engine.rs

```diff
@@ -524,7 +553,25 @@ fn apply_remote_items(db: &PlacesDb, scope: &SqlInterruptScope, now: Timestamp)
         root_guid = BookmarkRootGuid::Root.as_guid().as_str(),
         type_fragment = ItemTypeFragment("newKind"),
         sync_status = SyncStatus::Normal as u8,
-    ))?;
+    );
+
+    scope.err_if_interrupted()?;
+    let result = db.execute_batch(&upsert_sql);
+
+    // In trying to debug bug 1935797 - relaxing the trigger caused a spike on
+    // guid collisions, we want to report on this error before the actual upsert to see
+    // if we can discern any obvious signs
+    if let Err(rusqlite::Error::SqliteFailure(e, _)) = &result {
+        if e.code == ErrorCode::ConstraintViolation {
+            // This is a real collision that the sync code couldn't unify or fix.
+            error_support::report_error!(
+                "places-sync-bookmarks-guid-collision",
+                "GUID collision during bookmark sync upsert"
```

Comment:
(ie, see comment above, maybe here is also a good place to check if items remained orphaned?)

If you do decide to do this, maybe we just want a single log/report_error, reporting both the "before" and "after" values?

Reply:
I did end up adding an "after" for this but that had to be in the method higher up `update_local_items_in_places` where the triggers:
```
db.execute_batch("DELETE FROM changeGuidOps");
...
db.execute_batch("DELETE FROM applyNewLocalStructureOps")?;
```
run. That way we definitely know at that point dogear has fixed up what it could.